src/client.js

```diff
@@ -84,9 +92,27 @@ export default class StitchClient {
       authOptions.codec = options.authCodec;
     }
 
-    this.auth = new Auth(this, this.authUrl, authOptions);
-    this.auth.handleRedirect();
-    this.auth.handleCookie();
+    const authFactory = AuthFactory.create(stitchClient, stitchClient.authUrl, authOptions);
+    authFactory.then(auth => {
+      stitchClient.auth = auth;
+      stitchClient.auth.handleRedirect();
+      stitchClient.auth.handleCookie();
+
+      resolve(stitchClient);
+    });
+  });
+}
+/**
+ * Create a new _StitchClient instance.
```

Comment:
This was also not addressed from last comment. This docstring is incorrect and we are exposing the client. It's just not constructible.

Reply:
The docstring is wrong, but we discussed how we still need to expose the client internally.